samples/detect_intent_knowledge.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Dialogflow API Detect Knowledge Base Intent Python sample with text inputs.
+
+Examples:
+  python detect_intent_knowledge.py -h
+  python detect_intent_knowledge.py --project-id PROJECT_ID \
+  --session-id SESSION_ID --knowledge-base-id KNOWLEDGE_BASE_ID \
+  "hello" "how do I reset my password?"
+"""
+
+import argparse
+import uuid
+
+
+# [START dialogflow_detect_intent_knowledge]
+def detect_intent_knowledge(project_id, session_id, language_code,
+                            knowledge_base_id, texts):
+    """Returns the result of detect intent with querying Knowledge Connector.
+
+    Args:
+    project_id: The GCP project linked with the agent you are going to query.
+    session_id: Id of the session, using the same `session_id` between requests
+              allows continuation of the conversation.
+    language_code: Language of the queries.
+    knowledge_base_id: The Knowledge base's id to query against.
+    texts: A list of text queries to send.
+    """
+    import dialogflow_v2beta1 as dialogflow
+    session_client = dialogflow.SessionsClient()
+
+    session = session_client.session_path(project_id, session_id)
+    print('Session path: {}\n'.format(session))
+
+    for text in texts:
+        text_input = dialogflow.types.TextInput(
+            text=text, language_code=language_code)
+
+        query_input = dialogflow.types.QueryInput(text=text_input)
+
+        knowledge_base_name = dialogflow.knowledge_bases_client \
```

Comment:
I named it `knowledge_base_name` because line 60's QueryParameters Input is `knowledge_base_names` 

Bit of an inconsistency on the API, preference for which to use?